apps/homepage/__tests__/e2e/testMenu.cy.tsx

```diff
@@ -0,0 +1,65 @@
+describe('First test', () => {
+  it('Should check menu buttons see through out pages with lang en', () => {
+    cy.visit('/', {
+      headers: {
+        'Accept-Language': 'en'
+      }
+    });
+
+    cy.findByRole('link', {name: 'Home'}).should('exist').and('be.visible');
+    cy.findByRole('link', {name: 'Projects'}).should('exist').and('be.visible');
+    cy.findByRole('link', {name: 'Contact'}).should('exist').and('be.visible');
+
+    cy.visit('/projects', {
```

Comment:
I wouldn't test for the menu in all pages, I would have different test suites for testing different parts of the application:

```
- menu.cy.tsx
  - This file can test parts specific for the menu like it includes 3 options, with links, etc... 
- header.cy.tsx
  - This file can test that the header component includes the menu.
- homepage.cy.tsx
  - This file can test that the home page includes the header
- projects.cy.tsx
  - This file can test that the projects page includes the header
- contact.cy.tsx
  - This file can test that the contact page includes the header
```

Reply:
Basically you need to separete concerns and build test pages to serve a specific purpose.

Reply:
You could also create multiple folders like:

```
- components/
  - header.cy.tsx
  - menu.cy.tsx
- pages/
  - homepage.cy.tex
  - projects.cy.tex
  - contact.cy.tsx
```